src/storage/compression/bitpacking.cpp

```diff
@@ -179,12 +222,14 @@ struct BitpackingCompressState : public CompressionState {
 		}
 	}
 
-	void WriteValues(T *values, bitpacking_width_t width, idx_t count) {
+	void WriteValues(T *values, bitpacking_width_t width, T frame_of_reference, idx_t count) {
 		// TODO we can optimize this by stopping early if count < BITPACKING_WIDTH_GROUP_SIZE
 		BitpackingPrimitives::PackBuffer<T, false>(data_ptr, values, count, width);
 		data_ptr += (BITPACKING_WIDTH_GROUP_SIZE * width) / 8;
 
 		Store<bitpacking_width_t>(width, width_ptr);
+		width_ptr -= sizeof(T);
```

Comment:
width_ptr should be renamed probably as it no longer points to just widths, but to width + for.

Reply:
would `metadata_ptr` work?

Reply:
sounds good to me!